From 8f28a63f884c8e486c8edb079750680ec8238d95 Mon Sep 17 00:00:00 2001
From: "Cao,Bing Bu" <bingbu.cao@intel.com>
Date: Tue, 5 May 2015 13:57:27 +0800
Subject: [PATCH 397/462] atomisp2: fix kw issues in atomisp driver

Change-Id: I5a0589979139b3a6dfb500481effcedb2663c3dc
Tracked-On: #JIMINAN-35074
Signed-off-by: Cao,Bing Bu <bingbu.cao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/54283
Build: IMIN (SC)
Reviewed-by: Gang A Hu <gang.a.hu@intel.com>
Reviewed-by: Liang L Yang <liang.l.yang@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_cmd.c     | 17 +++++++++--------
 .../media/pci/atomisp2/atomisp_driver/atomisp_fops.c    |  2 +-
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 3de7a44..4572e6e 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -5217,6 +5217,7 @@ int atomisp_set_fmt(struct video_device *vdev, struct v4l2_format *f)
 	struct atomisp_video_pipe *pipe = atomisp_to_video_pipe(vdev);
 	struct atomisp_sub_device *asd = pipe->asd;
 	const struct atomisp_format_bridge *format_bridge;
+	const struct atomisp_format_bridge *snr_format_bridge;
 	struct atomisp_css_frame_info output_info, raw_output_info;
 	struct v4l2_format snr_fmt = *f;
 	struct v4l2_format backup_fmt = *f, s_fmt = *f;
@@ -5235,6 +5236,9 @@ int atomisp_set_fmt(struct video_device *vdev, struct v4l2_format *f)
 		f->fmt.pix.width, f->fmt.pix.height, source_pad,
 		asd->index, f->fmt.pix.bytesperline);
 
+	if (source_pad >= ATOMISP_SUBDEV_PADS_NUM)
+		return -EINVAL;
+
 	if (asd->streaming == ATOMISP_DEVICE_STREAMING_ENABLED) {
 		dev_warn(isp->dev, "ISP does not support set format while at streaming!\n");
 		return -EBUSY;
@@ -5391,24 +5395,21 @@ int atomisp_set_fmt(struct video_device *vdev, struct v4l2_format *f)
 	f->fmt.pix.width = snr_fmt.fmt.pix.width;
 	f->fmt.pix.height = snr_fmt.fmt.pix.height;
 
-	if (!atomisp_get_format_bridge(snr_fmt.fmt.pix.pixelformat))
+	snr_format_bridge =
+		atomisp_get_format_bridge(snr_fmt.fmt.pix.pixelformat);
+	if (!snr_format_bridge)
 		return -EINVAL;
 
 	atomisp_subdev_get_ffmt(&asd->subdev, NULL,
 				V4L2_SUBDEV_FORMAT_ACTIVE,
 				ATOMISP_SUBDEV_PAD_SINK)->code =
-		atomisp_get_format_bridge(
-			snr_fmt.fmt.pix.pixelformat)->mbus_code;
+		snr_format_bridge->mbus_code;
 
 	isp_sink_fmt = *atomisp_subdev_get_ffmt(&asd->subdev, NULL,
 					    V4L2_SUBDEV_FORMAT_ACTIVE,
 					    ATOMISP_SUBDEV_PAD_SINK);
 
-	if (!atomisp_get_format_bridge(f->fmt.pix.pixelformat))
-		return -EINVAL;
-
-	isp_source_fmt.code = atomisp_get_format_bridge(
-		f->fmt.pix.pixelformat)->mbus_code;
+	isp_source_fmt.code = format_bridge->mbus_code;
 	atomisp_subdev_set_ffmt(&asd->subdev, &fh,
 				V4L2_SUBDEV_FORMAT_ACTIVE,
 				source_pad, &isp_source_fmt);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index 7423122..4c933b2 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -225,7 +225,7 @@ int atomisp_q_video_buffers_to_css(struct atomisp_sub_device *asd,
 	struct atomisp_css_dvs_grid_info *dvs_grid =
 		 atomisp_css_get_dvs_grid_info(&asd->params.curr_grid_info);
 	unsigned long irqflags;
-	int err;
+	int err = 0;
 
 	while (pipe->buffers_in_css < ATOMISP_CSS_Q_DEPTH) {
 		struct videobuf_buffer *vb;
-- 
1.9.1

